addons/web_editor/static/src/js/editor/odoo-editor/src/utils/utils.js

```diff
@@ -1868,6 +1868,11 @@ export function splitElement(element, offset) {
         index < offset ? before.appendChild(child) : after.appendChild(child);
         index++;
     }
+    // e.g.: <p>Test/banner</p> + ENTER <=> <p>Test</p><div class="o_editor_banner>...</div><p><br></p>
+    const blockEl = closestBlock(after);
+    if (isShrunkBlock(blockEl)) {
```

Comment:
`fillEmpty` tests this too. Do you need to do it here as well so as not to insert zws? Otherwise you can remove this `if`.

Reply:
when I initially added the `fillEmpty` function, some test cases failed during local testing due to `zws` but now its working correctly. Therefore, it can be removed.

Reply:
I checked and there is no reason we should ever enter the `if` that adds `zws` in this case since we're passing it a block and if it's shrunk it will add a `<br>`, making it visible. If it's not shrunk, there's no reason it would not be visible. So this change is good, thanks.